cli/command/container/run.go

```diff
@@ -209,7 +205,7 @@ func runRun(dockerCli *command.DockerCli, flags *pflag.FlagSet, opts *runOptions
 		})
 	}
 
-	statusChan := waitExitOrRemoved(ctx, dockerCli, createResponse.ID, hostConfig.AutoRemove)
+	statusChan := waitExitOrRemoved(ctx, dockerCli, createResponse.ID, copts.autoRemove)
```

Comment:
> Even though that doesn't affect the code here

So, combined with the fact that `(*Client) ContainerCreate()` now changes the `hostConfig`, this *actually does* affect the code here.